src/components/FilterBar/FilterBar.test.tsx

```diff
@@ -0,0 +1,237 @@
+    ],
+  },
+  {
+    id: "tools",
+    name: "Tools",
+    items: [
+      { id: "hammer", name: "Hammer" },
+      {
+        id: "skrewdriver",
+        name: "Skrewdriver",
+        children: [
+          { id: "slottet", name: "Slotted Skrewdriver" },
+          { id: "phillips", name: "Phillips Skrewdriver" },
+          { id: "allen", name: "Allen Skrewdriver" },
+        ],
+      },
+      { id: "whisk", name: "Whisk" },
+    ],
+  },
+];
+
+const FilterBarTestComponent = () => {
+  const {
+    onChange,
+    onMixedStateChange,
+    selectedItems,
+    onClear,
+    onClearAll,
+    isModalOpen,
+    onToggle,
+  } = useFilterBar();
+  return (
+    <FilterBar
+      id="filterbar-id"
+      headingText="FilterBar"
+      layout="row"
+      isOpen={isModalOpen}
+      onToggle={onToggle}
+      onClear={onClearAll}
+      onSubmit={() => null}
+      selectedItems={selectedItems}
+      showClearAll={true}
+      showSubmitAll={true}
+    >
+      <MultiSelectGroup
+        id="multiselect-group"
+        key="multiselect-group-key"
+        labelText="MultiSelect Group"
+      >
+        {multiSelectItems &&
+          multiSelectItems.map((multiSelect) => (
+            <MultiSelect
+              key={multiSelect.id}
+              id={multiSelect.id}
+              label={multiSelect.name}
+              variant="dialog"
+              items={multiSelect.items}
+              selectedItems={selectedItems}
+              onChange={(e) => {
+                onChange(e.target.id, multiSelect.id);
+              }}
+              onMixedStateChange={(e) => {
+                onMixedStateChange(
+                  e.target.id,
+                  multiSelect.id,
+                  multiSelect.items
+                );
+              }}
+              onClear={() => {
+                onClear(multiSelect.id);
+              }}
+              onApply={() => {}}
+            />
+          ))}
+      </MultiSelectGroup>
+    </FilterBar>
+  );
+};
+const MulitSelectTestGroup = (multiSelectItems) => (
+  <MultiSelectGroup
+    id="MultiSelectGroup"
+    labelText="MultiSelectGroup example"
+    showLabel={true}
+    multiSelectWidth="default"
+  >
+    {multiSelectItems.map((multiSelectItem) => (
+      <MultiSelect
+        key={multiSelectItem.id}
+        id={multiSelectItem.id}
+        variant="listbox"
+        label={multiSelectItem.name}
+        items={multiSelectItem.items}
+        selectedItems={{}}
+        onChange={() => null}
+        onMixedStateChange={() => null}
+        onApply={() => {}}
+        onClear={() => "clear"}
+      />
+    ))}
+  </MultiSelectGroup>
+);
+describe("FilterBar Accessibility", () => {
+  it("should have no axe violations", async () => {
+    const { container } = render(<FilterBarTestComponent />);
+    expect(await axe(container)).toHaveNoViolations();
+  });
+});
+describe("FilterBar", () => {
+  it("should clear all selectedItems when 'Clear All' button is clicked", () => {
```

Comment:
There should be a test for the "Apply Filters" button that appears next to the "Clear filters" button.

Reply:
Added a bunch of tests...had some struggles mocking the viewport size. Now it is working but I am very aware that there might be a better way to do it. If you have an idea, pls point me in the right direction.